.github/workflows/ci.yml

```diff
@@ -98,10 +98,40 @@ jobs:
       - uses: Swatinem/rust-cache@v2.0.0
       - name: Install nextest
         uses: taiki-e/install-action@nextest
-      - name: Compile tests before running
-        run: cargo test --workspace --no-run
-      - name: Test execution
-        run: cargo nextest run --profile ci
+      - name: Build and archive tests
+        run: cargo nextest archive --archive-file nextest-archive.tar.zst
+      - name: Upload archive to workflow
+        uses: actions/upload-artifact@v3
+        with:
+          name: nextest-archive-${{ matrix.os }}
+          path: nextest-archive.tar.zst
+
+  run-tests-partitioned:
```

Comment:
❓ Do we need a new name for this?

I would have opted for the shorter `build-test` and `test` (or `run-test`).

Reply:
changed to `build-tests` and `run-tests`